### PR TITLE
tidyup kdbg of utils

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. Rename kdbg to utils
  It is difficult to know what kdbg is. Let's use system or utils instead of kdbg.
2. Remove cmd postfix from functions and configurations
  Each system util function can be used without TASH by function call.
  "CMD" can give confusion that it is a part of TASH command.
  CONFIG_SYSTEM_CMDS just registers enabled utils into TASH. It does not
  make them enabled.
  To use TASH commands of system utils like ps, free and etc, follow belows:
  1) enables it through CONFIG_ENABLE_<UTIL-NAME>
  2) enables CONFIG_SYSTEM_CMDS to use it from TASH
3. Remove CONFIG_DISABLE_ENVIRON conditional from env utils
  CONFIG_ENABLE_ENV_XX already has a dependency with CONFIG_DISABLE_ENVIRON
  so that we don't need to check it again at Makefile and source files.
4. Remove CONFIG_TASH conditional from utils.h
  System, filesystem and network commands configurations already have
  a dependency with CONFIG_TASH so that CONFIG_TASH is redundant in source code.
5. apps/utils: move prototypes of util functions to apps/include/system/utils.h
6. apps/init: remove unnecessary CONFIG_BUILD_PROTECTED conditional

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [bc871f3588df4c0fe6fd06af0c676043ef238e28] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [bc871f3588df4c0fe6fd06af0c676043ef238e28] - Code Rule Check OK.

File: apps/system/utils/systemcmd.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#if defined(CONFIG_ENABLE_PS)
+	{"ps",       utils_ps,           TASH_EXECMD_SYNC},
+#endif
+#if defined(CONFIG_ENABLE_ENV_SET)
+	{"setenv",   utils_env_set,      TASH_EXECMD_SYNC},
+#endif
+#if defined(CONFIG_ENABLE_STACKMONITOR)
+	{"stkmon",   utils_stackmonitor, TASH_EXECMD_SYNC},
+#endif
+#if defined(CONFIG_TTRACE)
```
Comment (jsdosa) [inline apps/system/utils/systemcmd.c]: How about adding "enable" to this?

**Comment by seinfra** [comment]
Target : [21da2ab53ac715b85ee7a74c010ffd55bc9e10d5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [21da2ab53ac715b85ee7a74c010ffd55bc9e10d5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [07211fdd924a11924547ccd84ad5c78dd2b4acbf] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [07211fdd924a11924547ccd84ad5c78dd2b4acbf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f779e5b643a8e65205fe115e3000f153e96a540d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f779e5b643a8e65205fe115e3000f153e96a540d] - Code Rule Check (C++) OK.

File: apps/system/utils/systemcmd.c (hunk 1 above)
Comment (sunghan-chang) [reply apps/system/utils/systemcmd.c]: That is not a part of this change. I will request this to the person who is in charge of ttrace.

**Pull request merged**